### ISSUE TITLE
shortcut Shift+Delete deletes instead of cutting to clipboard

### ISSUE THREAD

**Issue by bensoibj** [spyder-ide/spyder#5035]
## Description
The shortcut Shift+Delete used to be an alternative for Ctrl+X, cutting marked text to the clipboard. Since Spyder version 3.2 it only deletes the text, but does not copy it to the clipboard anymore. At least in the editor, in the console it works as expected.

**What steps will reproduce the problem?**

1. Mark some text in the editor
2. Cut the text using the shortcut Shift+Delete
3. Insert the text using Shift+Insert or Ctrl+V

**What is the expected output? What do you see instead?**
Some text is inserted, but it is not the cutted text, rather some previously copied text from another program, which is still in the clipboard. Step 3 works as expected, only if the shortcut Ctrl+X has been used for cutting.

**Please provide any additional information below**
In the console the shortcut works as expected.

## Version and main components

* Spyder Version: 3.2.1 
* Python Version: 3.6.2
* Qt Versions: 5.8.0, PyQt5 5.8.2 on Windows

## Dependencies
```
IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  0.26 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
numpy >=1.7      :  1.13.1 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.2 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
sympy >=0.7.3    :  1.1.1 (OK)

```

**Comment by ccordoba12** [comment]
Sorry, many users asked for Shift+Del to delete lines in the Editor. This is because many other editors and IDEs have the same behavior.

**Comment by bensoibj** [comment]
It's a pity that this cannot be configured in Preferences / Keyboard shortcuts, there only Ctrl+D appears as a shortcut to delete lines. Having never used Shift+Del for this, I have not yet adapted to it and still accidently delete lines... :-/

**Comment by ccordoba12** [comment]
Sorry for the inconvenience. When we were implementing `Shift+Del` we discovered it has to be hard-coded in the application to work correctly, so it can't be reconfigured to something else.

**Comment by prof7bit** [comment]
Can you please reopen this bug as a reminder that a way should be found to make it bahave like other editors too (shift-del = cut to clipboard)

> Sorry, many users asked for Shift+Del to delete lines in the Editor.

I don't believe this

> This is because many other editors and IDEs have the same behavior.

I think you must be confusing this with some other shortcut, virtually all Editors out there are using Shift+Del to cut, from textareas in web browsers to MS word, from geany to VS-Code, from Eclipse to Netbeans, the problem is since I am so used to it (it works everywhere and is the fastest way to cut & paste with keyboard) I regularly find myself accidentally deleting text instead of cutting whenever I use Spyder.

Please re-open this bug. You don't need to make it configurable, just make it behave correctly (as in how it works everywhere else too)

At least leave the bug opena as a reminder that it should be fixed eventually.

**Comment by ccordoba12** [comment]
> Can you please reopen this bug as a reminder that a way should be found to make it bahave like other editors too (shift-del = cut to clipboard)

It's not possible. `Shitf+Del` can't be made configurable because it needs to be hardcoded to work.

> I don't believe this

We tested it in Sublime and VSCode and Atom, the most popular editors right now.

**Comment by ccordoba12** [comment]
> At least leave the bug opena as a reminder that it should be fixed eventually.

We can't fix it, as I said above, sorry.

**Comment by prof7bit** [comment]
Strange. I'm using vs-code myself a lot (its my main editor nowadays) its doing clipboard-cut when pressing shift-del. Also atom and sublime did this when I used them. The same is true for PyCharm, Eclipse, Netbeans, IntelliJ, gedit, Geany, Kate, Notepad++, SciTE, MIcrosoft Word, LibreOffice, in fact I have not ever seen any editor or word processor that did not clipboard cut on shift-del. It's such a common shortcut, its defined here: https://en.wikipedia.org/wiki/IBM_Common_User_Access

**Comment by gherold** [comment]
As I also find the new behaviour extremely annoying, I want to add my voice to the plea to revert this change -- at least until a way is found to make the shortcut configurable.
Having used atom and sublime as well, I can't remember running into this problem earlier. However, it may be that these are recent changes in the other editors also (I hope they don't start defending their choice with "but it's how spyder does it")

**Comment by arneewald** [comment]
In Atom, for example, when pressing "shift-del", the corresponding line is cut to clipboard, i.e. the whole line is deleted and copied to clipboard. Furthermore, if text is selected, usually shift-del behaves like CTRL-X.

I would also very much appreciate adopting the standard behavior.

**Comment by EFG2** [comment]
Same here. Have been programming for the last 30 years, mainly on Windows. I'm so used to Shift+Del for cutting the selected text to clipboard that it's hard to replace by say 'Ctrl+X'. Visual Studio, Eclipse, Android Studio, Notepad++, FlashDevelop, even PyCharm on Linux accept Shift+Del and Shift+Insert for cutting and pasting of the selected text. I've seen the request for this new behavior but please do a poll to see how big the opposing parties actually are.

**Comment by ccordoba12** [comment]
This is already fixed in our latest version (3.2.6). Please update.

**Comment by Nils-Berghs** [comment]
Shift+Delete = CUT, end of discusion.

As a left handed mouse user I feel very strongly about this. Applications that do not treat Shift+Delete should be flagged as discriminating, and yes that includes Windows.
